src/Elastic.Markdown/IO/State/LinkReference.cs

```diff
@@ -15,8 +15,9 @@ public record LinkReference
 	[JsonPropertyName("url_path_prefix")]
 	public required string? UrlPathPrefix { get; init; }
 
+	/// Mapping of relative filepath and all the page's anchors for deeplinks
 	[JsonPropertyName("links")]
-	public required string[] Links { get; init; } = [];
+	public required Dictionary<string, string[]> Links { get; init; } = [];
 
 	[JsonPropertyName("cross_links")]
 	public required string[] CrossLinks { get; init; } = [];
```

Comment:
Do you think CrossLinks should have the same format? (Dictionary)

Reply:
Lets see if that becomes evident when we start working on resolving crosslinks.